src/devices/src/virtio/vsock/csm/connection.rs

```diff
@@ -443,6 +451,7 @@ where
                     0
                 });
             self.fwd_cnt += Wrapping(flushed as u32);
+            METRICS.vsock.tx_bytes_count.add(flushed as usize);
```

Comment:
I am afraid with this we are counting `tx_bytes` twice. The `tx_buffer` gets flushed to `self.stream` in `notify` and then the self.stream is written in `send_bytes`. So, we are counting the bytes twice. Am I wrong?

Reply:
The execution paths are different, but I can add unit tests to make sure that we don't actually count them twice.

Reply:
Unit test added.